src/StreetNameRegistry/Municipality/Events/StreetNameWasProposedV2.cs

```diff
@@ -56,9 +56,9 @@ ProvenanceData provenance
                 new NisCode(nisCode),
                 new Names(streetNameNames),
                 new PersistentLocalId(persistentLocalId))
-        => ((ISetProvenance)this).SetProvenance(provenance.ToProvenance());
+        => SetProvenance(provenance.ToProvenance());
 
-        void ISetProvenance.SetProvenance(Provenance provenance) => Provenance = new ProvenanceData(provenance);
+        public void SetProvenance(Provenance provenance) => Provenance = new ProvenanceData(provenance);
```

Comment:
Its actually better than the previous because it doesn't require a cast to ISetProvenance... sure its not entirely consistent now  but it is definitely better in use so maybe we can decide to implement it later?

Reply:
I also took this route because It was already present in a couple of other events, I saw that ISetProvenance was greyed out and then saw this implementation and decided to go with it.